client.go

```diff
@@ -167,6 +168,14 @@ func startContainer(img string, cmd []string, streaming, wait bool) (string, err
 		}
 	case <-timeout:
 		return id, fmt.Errorf("Timed out when waiting for a container.\n")
+	case <-killChannel:
+		log.Println("going to kill")
+		if err := client.KillContainer(id, "KILL"); err != nil {
+			fmt.Println("Error while killing running container:", err)
```

Comment:
Yep, I'm going to fix it
On 28 Jul 2015 20:13, "Miquel Sabaté Solà" <notifications@github.com> wrote:

> In client.go
> <https://github.com/SUSE/zypper-docker/pull/19#discussion_r35680513>:
>
> > @@ -167,6 +168,14 @@ func startContainer(img string, cmd []string, streaming, wait bool) (string, err
> >  		}
> >  	case <-timeout:
> >  		return id, fmt.Errorf("Timed out when waiting for a container.\n")
> > +	case <-killChannel:
> > +		log.Println("going to kill")
> > +		if err := client.KillContainer(id, "KILL"); err != nil {
> > +			fmt.Println("Error while killing running container:", err)
>
> Shouldn't we exitWithCode(1) in this case?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SUSE/zypper-docker/pull/19/files#r35680513>.
>